### PR TITLE
[INLONG-3243][Sort-Standalone] Support multiple scenes to request configs

### LINKED ISSUE #3243: [Feature][Sort-Standalone] Support multiple scenes to request configs
### Description

support three scenes to request config
1. by fixed manager address (Default, from CommonPropertiesHolder)
2. by name service and dynamic manager address (plugin)
3. by local file or database (plugin)

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
### Title Name: [INLONG-3243][Sort-Standalone] Support multiple scenes to request configs

Fixes #3243 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/SortSdkSource.java [hunk 1]
```diff
@@ -207,7 +220,7 @@ private void updateAllClientConfig() {
      * @param config The config to be updated.
      */
     private void updateClientConfig(SortClientConfig config) {
-        config.setManagerApiUrl(CommonPropertiesHolder.getSourceConfigManagerUrl());
+        config.setManagerApiUrl(ManagerAddrGetHandler.getSortSourceConfigUrl());
```
Comment (luchunliang) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/SortSdkSource.java:223]: When common.properties have the key "sortSourceConfig.QueryConsumeConfigType", the method "updateClientConfig" is unusable.

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/CommonPropertiesHolder.java [hunk 2]
```diff
@@ -196,22 +200,22 @@ public static Integer getInteger(String key) {
...
     /**
-     * getAuditFormatInterval
-     * 
+     * get if managerAddrNeedUpdate
+     *
      * @return
      */
-    public static long getAuditFormatInterval() {
-        return auditFormatInterval;
+    public static boolean isManagerAddrNeedUpdate() {
```
Comment (luchunliang) [inline inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/CommonPropertiesHolder.java:218]: The method "isManagerAddrNeedUpdate" is unusable.

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/ManagerAddrGetHandler.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+import java.util.Optional;
+
+/**
+ * Manager address get handler.
+ *
+ * <p> Used to acquire the ip and port of manager, which sort sdk and sort-standalone request config from. </p>
+ * <p> The default implementation is base on {@link CommonPropertiesHolder} to acquire properties. </p>
+ */
+public class ManagerAddrGetHandler implements Configurable {
```
Comment (luchunliang) [inline inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/ManagerAddrGetHandler.java:35]: "Addr" is not good concept.
ManagerUrlHandler is better.

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/ManagerAddrGetHandler.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    }
+
+    /**
+     * Get URL where SortSdk request SortSourceConfig.
+     *
+     * @return URL to get SortSourceConfig.
+     */
+    public static String getSortSourceConfigUrl() {
+        String url = get().context.getString(KEY_SORT_SOURCE_CONFIG_MANAGER_URL);
```
Comment (luchunliang) [inline inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/ManagerAddrGetHandler.java:59]: It is not good to get url from context every time.

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/ManagerAddrGetHandler.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        return url;
+    }
+
+    /**
+     * Get URL where Sort-Standalone request SortClusterConfig.
+     *
+     * @return URL to get SortClusterConfig.
+     */
+    public static String getSortClusterConfigUrl() {
+        String url = get().context.getString(KEY_SORT_CLUSTER_CONFIG_MANAGER_URL);
```
Comment (luchunliang) [inline inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/ManagerAddrGetHandler.java:73]: ditto

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/CommonPropertiesHolder.java (hunk 2 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/CommonPropertiesHolder.java:218]: maybe this managerAddrNeedUpdate is useless, need further disscuss

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/SortSdkSource.java (hunk 1 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/SortSdkSource.java:223]: in line 243, updateClientConfig is used before init sort client

**Review by luchunliang — APPROVED** [review]

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/ManagerAddrGetHandler.java (hunk 3 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/ManagerAddrGetHandler.java:35]: fixed

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/ManagerAddrGetHandler.java (hunk 4 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/ManagerAddrGetHandler.java:59]: fix, use local cache, only get it in the first time

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/ManagerAddrGetHandler.java (hunk 5 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/ManagerAddrGetHandler.java:73]: fixed

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**